task_force_bot.py

```diff
@@ -78,158 +67,496 @@ def distance_1d(val_1: Union[int, np.ndarray], val_2: Union[int, np.ndarray], si
+            return rng.choice(np.array(['NORTH', 'EAST', 'SOUTH', 'WEST']))
+
+    def task_build_shipyard(self, ship_position: int) -> str:
+        shipyard_position = np.nan
+        for shipyard in self.shipyards:
+            shipyard_position = shipyard.position
+            if shipyard.planned:
+                break
+
+        if ship_position == shipyard_position:
+            return 'CONVERT'
+        return self.pathfinder(ship_position, shipyard_position)
+
+    def find_closest_shipyard_position(self, ship_position: int, shipyard_positions: list) -> int:
+        closest_shipyard_position = np.nan
+        closest_shipyard_distance = np.inf
+        for shipyard_position in shipyard_positions:
+            if self.distance_matrix[ship_position, shipyard_position] < closest_shipyard_distance:
+                closest_shipyard_position = shipyard_position
+                closest_shipyard_distance = self.distance_matrix[ship_position, shipyard_position]
+
+        return closest_shipyard_position
+
+    def task_return_halite(self, ship_position: int) -> str:
+        existing_shipyards_positions = []
+        for shipyard in self.shipyards:
+            if shipyard.exists:
+                existing_shipyards_positions.append(shipyard.position)
+
+        closest_shipyard_position = self.find_closest_shipyard_position(ship_position, existing_shipyards_positions)
+
+        return self.pathfinder(ship_position, closest_shipyard_position)
+
+    # Pathfinder #######################################################################################################
+
+    def pathfinder(self, current_position: int, target_position: int) -> Union[str, None]:
+        current_coordinates = get_coordinates(current_position, self.size)
+
+        directions_values = {}
+        for direction in self.directions_dict:
+            directions_values[direction] = (self.distance_matrix[target_position].reshape(self.size, -1)
+                                            .item(tuple((current_coordinates
+                                                         + self.directions_dict[direction]) % self.size))
+                                            - self.distance_matrix[target_position].reshape(self.size, -1)
+                                            .item(tuple(current_coordinates)))
+
+        for direction in self.directions_dict:
+            if self.blocked_squares.item(
+                    tuple((current_coordinates + self.directions_dict[direction]) % self.size)):
+                directions_values.pop(direction, None)
+            # TODO - Improve on simply blocking all threatened squares
+            if self.board_threatened_squares.item(
+                    tuple((current_coordinates + self.directions_dict[direction]) % self.size)):
+                directions_values.pop(direction, None)
+
+        # For the rare occasion that all field are blocked use try and stay put if nothing is possible
+        try:
```

Comment:
lieber überprüfen, ob es werte in dem dict gibt. Wenn ja, nimm das Minimum. Wenn nein, gib nichts zurück